File: haesa/0x05/10828.cpp
#include <bits/stdc++.h>
using namespace std;
stack<int> s;
string cmd;
int n;
int k;
int main() {
  ios::sync_with_stdio(0);
  cin.tie(0);

  cin >> n;
  for (int i = 0; i < n; i++) {
    cin >> cmd;
    if (cmd == "push") {
      cin >> k;
      s.push(k);
    } else if (cmd == "pop") {
      if (s.empty())
        cout << -1 << '\n';
      else {
        cout << s.top() << '\n';
        s.pop();
      }
    } else if (cmd == "size")
      cout << s.size() << '\n';
    else if (cmd == "empty")
      cout << s.empty() << '\n';
    else if (cmd == "top") {
      if (s.empty())
        cout << -1 << '\n';
      else
        cout << s.top() << '\n';
    }
  }
}

File: haesa/0x05/10828.js
const input = require('fs')
  .readFileSync('/dev/stdin')
  .toString()
  .trim()
  .split('\n');

const n = Number(input[0]);
const stack = [];
const result = [];

for (let i = 1; i <= n; i++) {
  const [cmd, num] = input[i].trim().split(' ');
  if (cmd == 'push') stack.push(Number(num));
  else if (cmd == 'pop') result.push(stack.length ? stack.pop() : -1);
  else if (cmd == 'size') result.push(stack.length);
  else if (cmd == 'empty') result.push(stack.length ? 0 : 1);
  else if (cmd == 'top')
    result.push(stack.length ? stack[stack.length - 1] : -1);
}
console.log(...result);
